packages/app/src/components/nominationInfo/ApplicationForm.js

```diff
@@ -0,0 +1,311 @@
+import React, { useEffect, useRef, useContext, useState } from 'react';
+// import styles from './styles.module.css';
+import styles from './newstyles.module.css';
+import { useForm, Controller } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as Yup from 'yup';
+import nominationsAPI from '../../utils/API/nominationsAPI';
+import { NominationsDataContext } from '../../utils/context/NominationsContext';
+import { ActiveNominationContext } from '../../utils/context/ActiveNominationContext';
+import DatePicker, { CalendarContainer } from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+
+const ApplicationForm = props => {
+  const firstUpdate = useRef(true);
+
+  const [NominationsData, setNominationsData] = useContext(
+    NominationsDataContext
+  );
+
+  const [activeNomination, setActiveNomination] = useContext(
+    ActiveNominationContext
+  );
+
+  const [admissionDate, setAdmissionDate] = useState(new Date());
+  const [dischargeDate, setDischargeDate] = useState(new Date());
+
+  useEffect(() => {
+    if (props.formData['Admission Date'] != 'Invalid Date') {
+      setAdmissionDate(props.formData['Admission Date']);
+    }
+  }, [props.formData['Admission Date']]);
+
+  useEffect(() => {
+    if (props.formData['Discharge Date'] != 'Invalid Date') {
+      setDischargeDate(props.formData['Discharge Date']);
+    }
+  }, [props.formData['Discharge Date']]);
+
+  const phoneRegex = /^[\+]?[(]?[0-9]{3}[)]?[-\s\.]?[0-9]{3}[-\s\.]?[0-9]{4,6}$/;
+  const yesNoRegex = /^(?:Yes\b|No\b)/;
+
+  useEffect(() => {
+    if (!firstUpdate.current) {
+      handleSubmit(submitForm)();
+    }
+    firstUpdate.current = false;
+  }, [props.saveHasBeenClicked]);
+
+  const validationSchema = Yup.object({
+    'Admission Date': Yup.date().required('Required'),
+    'Discharge Date': Yup.date()
+      .min(
+        Yup.ref('Admission Date'),
+        'Discharge date cannot be before admission date.'
+      )
+      .required('Required'),
+    'Representative Name': Yup.string()
+      .min(3, 'Must be 3 characters or more.')
+      .max(30, 'Must be 30 characters or less.')
+      .required('Required'),
+    'Representative Email Address': Yup.string()
+      .email('Invalid email address.')
+      .required('Required'), // This handles email validation with no regex.
+    'Representative Phone Number': Yup.string()
```

Comment:
Instead of using a raw string with regex validation, maybe see if this phone number input package is worth checking out https://gitlab.com/catamphetamine/react-phone-number-input. The demo looks pretty nice and it would probably be more user-friendly since it has auto formatting.

Reply:
@t-keazirian @somersbmatthews let's find yup built in phone number formatting like for email.

Reply:
I saw a few yup things that might be useful for validation, but what's cool about that one is that it formats the input as the user types. Would be nice to have that but not a big deal if it's too complicated.

Reply:
@aschey the yup-phone we tried https://www.npmjs.com/package/yup-phone also validates whether or not the phone number actually exists in the real world, which is not workable with dummy data, so we decided to keep the regex.

Reply:
Did you check that other package I linked above?

Reply:
no, checking now

Reply:
@aschey fixed :
`    'Representative Phone Number': Yup.string()
      .test('test-name', 'must be valid phone number', 
        function(value) {
          return isValidPhoneNumber(`+1${value}`)
        })
      .required('Required'),`

Reply:
and added formatting from `react-phone-number-input`
```javascript
          nomination.representativePhoneNumber =
          formatPhoneNumber(`+1${data['Representative Phone Number']}`);
```

Reply:
Nice, looks good.